### PR TITLE
nice error msg from load_state_dict for non-tensor value

### LINKED ISSUE #67549: Strange nested exception during wrong call of load_state_dict
```
Traceback (most recent call last):
  File "/lab/vadim/prefix/miniconda/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1336, in _load_from_state_dict
    param.copy_(input_param)
TypeError: copy_(): argument 'other' (position 1) must be Tensor, not numpy.ndarray

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/lab/vadim/detregwsod/train.py", line 49, in <module>
    main(args)
  File "/lab/vadim/detregwsod/train.py", line 38, in main
    model.backbone.load_state_dict(state_dict)
  File "/lab/vadim/prefix/miniconda/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1392, in load_state_dict
    load(self)
  File "/lab/vadim/prefix/miniconda/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1390, in load
    load(child, prefix + name + '.')
  File "/lab/vadim/prefix/miniconda/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1390, in load
    load(child, prefix + name + '.')
  File "/lab/vadim/prefix/miniconda/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1386, in load
    module._load_from_state_dict(
  File "/lab/vadim/prefix/miniconda/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1342, in _load_from_state_dict
    .format(key, param.size(), input_param.size(), ex.args))
TypeError: 'int' object is not callable
```  

cc @albanD @mruberry @jbschlosser @walterddr

### PR DESCRIPTION
**Pull request by kshitij12345**
Fixes https://github.com/pytorch/pytorch/issues/67549

### CODE REVIEW

File: torch/nn/modules/module.py [hunk 1]
```diff
@@ -1387,6 +1387,13 @@ def _load_from_state_dict(self, state_dict, prefix, local_metadata, strict,
             key = prefix + name
             if key in state_dict:
                 input_param = state_dict[key]
+                if not torch.is_tensor(input_param):
```
Comment (albanD) [inline torch/nn/modules/module.py:1390]: Shouldn't this be https://pytorch.org/docs/stable/torch.overrides.html?highlight=is_tensor_like#torch.overrides.is_tensor_like ?

Comment (jbschlosser) [reply torch/nn/modules/module.py:1390]: This will become more important as we broaden support for custom tensors.

**Review by jbschlosser — APPROVED** [review]
LGTM! Thanks for the fix :)

**Comment by kshitij12345** [comment]
Gentle ping @jbschlosser

**Comment by kshitij12345** [comment]
Gentle ping @jbschlosser

**Comment by jbschlosser** [comment]
@kshitij12345 sorry, thought Alban was going to import this one :) thanks for the ping!

**Comment by vadimkantorov** [comment]
Is there anywhere in docs a definition / spec of a "tensor-like object"?

**Pull request closed without merging**